test/gold/allocation-sizes.clang.-g-grecord-gcc-switches-O1-O0-fno-discard-value-names.1-callsite.allocation_size.golden.csv

```diff
@@ -7,15 +7,12 @@
 *global_alloc@.str[*][*]	1
 *global_alloc@.str[0]	13
 *global_alloc@.str[0][*]	1
-*global_alloc@dynamic_size_global	8
 *global_alloc@dynamic_size_global[*]	8
 *global_alloc@dynamic_size_global[0]	8
-*global_alloc@large_const_size_global	512
 *global_alloc@large_const_size_global[*]	512
 *global_alloc@large_const_size_global[*][*]	4
 *global_alloc@large_const_size_global[0]	512
 *global_alloc@large_const_size_global[0][*]	4
-*global_alloc@small_const_size_global	32
 *global_alloc@small_const_size_global[*]	32
 *global_alloc@small_const_size_global[*][*]	4
 *global_alloc@small_const_size_global[0]	32
```

Comment:
Triage: These changes are suspicious and should be reviewed before merging

Reply:
```bash
alias docker-here='docker run --rm -it --mount type=bind,src=$PWD,target=/work --workdir /work'
docker-here cclyzer-dev-llvm-11 clang-11 -O1 -O0 -fno-discard-value-names -S -emit-llvm test/c/allocation-sizes.c -o 11.ll
docker pull ghcr.io/galoisinc/cclyzerpp-dev:main
docker tag ghcr.io/galoisinc/cclyzerpp-dev:main cclyzer-dev
docker-here cclyzer-dev clang-10 -O1 -O0 -fno-discard-value-names -S -emit-llvm test/c/allocation-sizes.c -o 10.ll
```

<details>
<summary>10.ll</summary>

```
; ModuleID = 'test/c/allocation-sizes.c'
source_filename = "test/c/allocation-sizes.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [13 x i8] c"DYNAMIC_DATA\00", align 1
@dynamic_size_global = common dso_local global i32* null, align 8
@small_const_size_global = common dso_local global [8 x i32] zeroinitializer, align 16
@large_const_size_global = common dso_local global [128 x i32] zeroinitializer, align 16

; Function Attrs: noinline nounwind optnone uwtable
define dso_local void @set(i32* %buf, i32 %sz) #0 {
entry:
  %buf.addr = alloca i32*, align 8
  %sz.addr = alloca i32, align 4
  store i32* %buf, i32** %buf.addr, align 8
  store i32 %sz, i32* %sz.addr, align 4
  %0 = load i32*, i32** %buf.addr, align 8
  %1 = bitcast i32* %0 to i8*
  %call = call i32 @rand() #4
  %2 = trunc i32 %call to i8
  %3 = load i32, i32* %sz.addr, align 4
  %conv = sext i32 %3 to i64
  call void @llvm.memset.p0i8.i64(i8* align 4 %1, i8 %2, i64 %conv, i1 false)
  %4 = load i32*, i32** %buf.addr, align 8
  %arrayidx = getelementptr inbounds i32, i32* %4, i64 1
  store i32 5, i32* %arrayidx, align 4
  ret void
}

; Function Attrs: nounwind
declare dso_local i32 @rand() #1

; Function Attrs: argmemonly nounwind willreturn
declare void @llvm.memset.p0i8.i64(i8* nocapture writeonly, i8, i64, i1 immarg) #2

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main() #0 {
entry:
  %retval = alloca i32, align 4
  %dyn = alloca i32, align 4
  %small_const_size_stack = alloca [8 x i32], align 16
  %large_const_size_stack = alloca [128 x i32], align 16
  %saved_stack = alloca i8*, align 8
  %__vla_expr0 = alloca i64, align 8
  %small_const_size_heap = alloca i32*, align 8
  %large_const_size_heap = alloca i32*, align 8
  %dynamic_size_heap = alloca i32*, align 8
  store i32 0, i32* %retval, align 4
  %call = call i8* @getenv(i8* getelementptr inbounds ([13 x i8], [13 x i8]* @.str, i64 0, i64 0)) #4
  %call1 = call i32 @atoi(i8* %call) #5
  store i32 %call1, i32* %dyn, align 4
  %0 = load i32, i32* %dyn, align 4
  %conv = sext i32 %0 to i64
  %mul = mul i64 4, %conv
  %call2 = call noalias i8* @malloc(i64 %mul) #4
  %1 = bitcast i8* %call2 to i32*
  store i32* %1, i32** @dynamic_size_global, align 8
  %2 = load i32, i32* %dyn, align 4
  %3 = zext i32 %2 to i64
  %4 = call i8* @llvm.stacksave()
  store i8* %4, i8** %saved_stack, align 8
  %vla = alloca i32, i64 %3, align 16
  store i64 %3, i64* %__vla_expr0, align 8
  %call3 = call noalias i8* @malloc(i64 32) #4
  %5 = bitcast i8* %call3 to i32*
  store i32* %5, i32** %small_const_size_heap, align 8
  %call4 = call noalias i8* @malloc(i64 512) #4
  %6 = bitcast i8* %call4 to i32*
  store i32* %6, i32** %large_const_size_heap, align 8
  %7 = load i32, i32* %dyn, align 4
  %conv5 = sext i32 %7 to i64
  %mul6 = mul i64 4, %conv5
  %call7 = call noalias i8* @malloc(i64 %mul6) #4
  %8 = bitcast i8* %call7 to i32*
  store i32* %8, i32** %dynamic_size_heap, align 8
  call void @set(i32* getelementptr inbounds ([8 x i32], [8 x i32]* @small_const_size_global, i64 0, i64 0), i32 8)
  call void @set(i32* getelementptr inbounds ([128 x i32], [128 x i32]* @large_const_size_global, i64 0, i64 0), i32 128)
  %9 = load i32*, i32** @dynamic_size_global, align 8
  %10 = load i32, i32* %dyn, align 4
  call void @set(i32* %9, i32 %10)
  %arraydecay = getelementptr inbounds [8 x i32], [8 x i32]* %small_const_size_stack, i64 0, i64 0
  call void @set(i32* %arraydecay, i32 8)
  %arraydecay8 = getelementptr inbounds [128 x i32], [128 x i32]* %large_const_size_stack, i64 0, i64 0
  call void @set(i32* %arraydecay8, i32 128)
  %11 = load i32, i32* %dyn, align 4
  call void @set(i32* %vla, i32 %11)
  %12 = load i32*, i32** %small_const_size_heap, align 8
  call void @set(i32* %12, i32 8)
  %13 = load i32*, i32** %large_const_size_heap, align 8
  call void @set(i32* %13, i32 128)
  %14 = load i32*, i32** %dynamic_size_heap, align 8
  %15 = load i32, i32* %dyn, align 4
  call void @set(i32* %14, i32 %15)
  %16 = load i32, i32* getelementptr inbounds ([8 x i32], [8 x i32]* @small_const_size_global, i64 0, i64 0), align 16
  %17 = load i32, i32* getelementptr inbounds ([128 x i32], [128 x i32]* @large_const_size_global, i64 0, i64 0), align 16
  %add = add nsw i32 %16, %17
  %18 = load i32*, i32** @dynamic_size_global, align 8
  %arrayidx = getelementptr inbounds i32, i32* %18, i64 0
  %19 = load i32, i32* %arrayidx, align 4
  %add9 = add nsw i32 %add, %19
  %arrayidx10 = getelementptr inbounds [8 x i32], [8 x i32]* %small_const_size_stack, i64 0, i64 0
  %20 = load i32, i32* %arrayidx10, align 16
  %add11 = add nsw i32 %add9, %20
  %arrayidx12 = getelementptr inbounds [128 x i32], [128 x i32]* %large_const_size_stack, i64 0, i64 0
  %21 = load i32, i32* %arrayidx12, align 16
  %add13 = add nsw i32 %add11, %21
  %arrayidx14 = getelementptr inbounds i32, i32* %vla, i64 0
  %22 = load i32, i32* %arrayidx14, align 16
  %add15 = add nsw i32 %add13, %22
  %23 = load i32*, i32** %small_const_size_heap, align 8
  %arrayidx16 = getelementptr inbounds i32, i32* %23, i64 0
  %24 = load i32, i32* %arrayidx16, align 4
  %add17 = add nsw i32 %add15, %24
  %25 = load i32*, i32** %large_const_size_heap, align 8
  %arrayidx18 = getelementptr inbounds i32, i32* %25, i64 0
  %26 = load i32, i32* %arrayidx18, align 4
  %add19 = add nsw i32 %add17, %26
  %27 = load i32*, i32** %dynamic_size_heap, align 8
  %arrayidx20 = getelementptr inbounds i32, i32* %27, i64 0
  %28 = load i32, i32* %arrayidx20, align 4
  %add21 = add nsw i32 %add19, %28
  store i32 %add21, i32* %retval, align 4
  %29 = load i8*, i8** %saved_stack, align 8
  call void @llvm.stackrestore(i8* %29)
  %30 = load i32, i32* %retval, align 4
  ret i32 %30
}

; Function Attrs: nounwind readonly
declare dso_local i32 @atoi(i8*) #3

; Function Attrs: nounwind
declare dso_local i8* @getenv(i8*) #1

; Function Attrs: nounwind
declare dso_local noalias i8* @malloc(i64) #1

; Function Attrs: nounwind
declare i8* @llvm.stacksave() #4

; Function Attrs: nounwind
declare void @llvm.stackrestore(i8*) #4

attributes #0 = { noinline nounwind optnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { nounwind "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { argmemonly nounwind willreturn }
attributes #3 = { nounwind readonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #4 = { nounwind }
attributes #5 = { nounwind readonly }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"clang version 10.0.0-4ubuntu1 "}
```

</details>

<details>
<summary>11.ll</summary>

```
; ModuleID = 'test/c/allocation-sizes.c'
source_filename = "test/c/allocation-sizes.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [13 x i8] c"DYNAMIC_DATA\00", align 1
@dynamic_size_global = dso_local global i32* null, align 8
@small_const_size_global = dso_local global [8 x i32] zeroinitializer, align 16
@large_const_size_global = dso_local global [128 x i32] zeroinitializer, align 16

; Function Attrs: noinline nounwind optnone uwtable
define dso_local void @set(i32* %buf, i32 %sz) #0 {
entry:
  %buf.addr = alloca i32*, align 8
  %sz.addr = alloca i32, align 4
  store i32* %buf, i32** %buf.addr, align 8
  store i32 %sz, i32* %sz.addr, align 4
  %0 = load i32*, i32** %buf.addr, align 8
  %1 = bitcast i32* %0 to i8*
  %call = call i32 @rand() #4
  %2 = trunc i32 %call to i8
  %3 = load i32, i32* %sz.addr, align 4
  %conv = sext i32 %3 to i64
  call void @llvm.memset.p0i8.i64(i8* align 4 %1, i8 %2, i64 %conv, i1 false)
  %4 = load i32*, i32** %buf.addr, align 8
  %arrayidx = getelementptr inbounds i32, i32* %4, i64 1
  store i32 5, i32* %arrayidx, align 4
  ret void
}

; Function Attrs: nounwind
declare dso_local i32 @rand() #1

; Function Attrs: argmemonly nounwind willreturn writeonly
declare void @llvm.memset.p0i8.i64(i8* nocapture writeonly, i8, i64, i1 immarg) #2

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main() #0 {
entry:
  %retval = alloca i32, align 4
  %dyn = alloca i32, align 4
  %small_const_size_stack = alloca [8 x i32], align 16
  %large_const_size_stack = alloca [128 x i32], align 16
  %saved_stack = alloca i8*, align 8
  %__vla_expr0 = alloca i64, align 8
  %small_const_size_heap = alloca i32*, align 8
  %large_const_size_heap = alloca i32*, align 8
  %dynamic_size_heap = alloca i32*, align 8
  store i32 0, i32* %retval, align 4
  %call = call i8* @getenv(i8* getelementptr inbounds ([13 x i8], [13 x i8]* @.str, i64 0, i64 0)) #4
  %call1 = call i32 @atoi(i8* %call) #5
  store i32 %call1, i32* %dyn, align 4
  %0 = load i32, i32* %dyn, align 4
  %conv = sext i32 %0 to i64
  %mul = mul i64 4, %conv
  %call2 = call noalias i8* @malloc(i64 %mul) #4
  %1 = bitcast i8* %call2 to i32*
  store i32* %1, i32** @dynamic_size_global, align 8
  %2 = load i32, i32* %dyn, align 4
  %3 = zext i32 %2 to i64
  %4 = call i8* @llvm.stacksave()
  store i8* %4, i8** %saved_stack, align 8
  %vla = alloca i32, i64 %3, align 16
  store i64 %3, i64* %__vla_expr0, align 8
  %call3 = call noalias i8* @malloc(i64 32) #4
  %5 = bitcast i8* %call3 to i32*
  store i32* %5, i32** %small_const_size_heap, align 8
  %call4 = call noalias i8* @malloc(i64 512) #4
  %6 = bitcast i8* %call4 to i32*
  store i32* %6, i32** %large_const_size_heap, align 8
  %7 = load i32, i32* %dyn, align 4
  %conv5 = sext i32 %7 to i64
  %mul6 = mul i64 4, %conv5
  %call7 = call noalias i8* @malloc(i64 %mul6) #4
  %8 = bitcast i8* %call7 to i32*
  store i32* %8, i32** %dynamic_size_heap, align 8
  call void @set(i32* getelementptr inbounds ([8 x i32], [8 x i32]* @small_const_size_global, i64 0, i64 0), i32 8)
  call void @set(i32* getelementptr inbounds ([128 x i32], [128 x i32]* @large_const_size_global, i64 0, i64 0), i32 128)
  %9 = load i32*, i32** @dynamic_size_global, align 8
  %10 = load i32, i32* %dyn, align 4
  call void @set(i32* %9, i32 %10)
  %arraydecay = getelementptr inbounds [8 x i32], [8 x i32]* %small_const_size_stack, i64 0, i64 0
  call void @set(i32* %arraydecay, i32 8)
  %arraydecay8 = getelementptr inbounds [128 x i32], [128 x i32]* %large_const_size_stack, i64 0, i64 0
  call void @set(i32* %arraydecay8, i32 128)
  %11 = load i32, i32* %dyn, align 4
  call void @set(i32* %vla, i32 %11)
  %12 = load i32*, i32** %small_const_size_heap, align 8
  call void @set(i32* %12, i32 8)
  %13 = load i32*, i32** %large_const_size_heap, align 8
  call void @set(i32* %13, i32 128)
  %14 = load i32*, i32** %dynamic_size_heap, align 8
  %15 = load i32, i32* %dyn, align 4
  call void @set(i32* %14, i32 %15)
  %16 = load i32, i32* getelementptr inbounds ([8 x i32], [8 x i32]* @small_const_size_global, i64 0, i64 0), align 16
  %17 = load i32, i32* getelementptr inbounds ([128 x i32], [128 x i32]* @large_const_size_global, i64 0, i64 0), align 16
  %add = add nsw i32 %16, %17
  %18 = load i32*, i32** @dynamic_size_global, align 8
  %arrayidx = getelementptr inbounds i32, i32* %18, i64 0
  %19 = load i32, i32* %arrayidx, align 4
  %add9 = add nsw i32 %add, %19
  %arrayidx10 = getelementptr inbounds [8 x i32], [8 x i32]* %small_const_size_stack, i64 0, i64 0
  %20 = load i32, i32* %arrayidx10, align 16
  %add11 = add nsw i32 %add9, %20
  %arrayidx12 = getelementptr inbounds [128 x i32], [128 x i32]* %large_const_size_stack, i64 0, i64 0
  %21 = load i32, i32* %arrayidx12, align 16
  %add13 = add nsw i32 %add11, %21
  %arrayidx14 = getelementptr inbounds i32, i32* %vla, i64 0
  %22 = load i32, i32* %arrayidx14, align 16
  %add15 = add nsw i32 %add13, %22
  %23 = load i32*, i32** %small_const_size_heap, align 8
  %arrayidx16 = getelementptr inbounds i32, i32* %23, i64 0
  %24 = load i32, i32* %arrayidx16, align 4
  %add17 = add nsw i32 %add15, %24
  %25 = load i32*, i32** %large_const_size_heap, align 8
  %arrayidx18 = getelementptr inbounds i32, i32* %25, i64 0
  %26 = load i32, i32* %arrayidx18, align 4
  %add19 = add nsw i32 %add17, %26
  %27 = load i32*, i32** %dynamic_size_heap, align 8
  %arrayidx20 = getelementptr inbounds i32, i32* %27, i64 0
  %28 = load i32, i32* %arrayidx20, align 4
  %add21 = add nsw i32 %add19, %28
  store i32 %add21, i32* %retval, align 4
  %29 = load i8*, i8** %saved_stack, align 8
  call void @llvm.stackrestore(i8* %29)
  %30 = load i32, i32* %retval, align 4
  ret i32 %30
}

; Function Attrs: nounwind readonly
declare dso_local i32 @atoi(i8*) #3

; Function Attrs: nounwind
declare dso_local i8* @getenv(i8*) #1

; Function Attrs: nounwind
declare dso_local noalias i8* @malloc(i64) #1

; Function Attrs: nounwind
declare i8* @llvm.stacksave() #4

; Function Attrs: nounwind
declare void @llvm.stackrestore(i8*) #4

attributes #0 = { noinline nounwind optnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #1 = { nounwind "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #2 = { argmemonly nounwind willreturn writeonly }
attributes #3 = { nounwind readonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
attributes #4 = { nounwind }
attributes #5 = { nounwind readonly }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
```

</details>

<details>
<summary> `diff -u 10.ll 11.ll` </summary>

```diff
--- 10.ll	2022-10-25 11:58:27.530693041 -0400
+++ 11.ll	2022-10-25 11:58:26.684687684 -0400
@@ -4,9 +4,9 @@
 target triple = "x86_64-pc-linux-gnu"
 
 @.str = private unnamed_addr constant [13 x i8] c"DYNAMIC_DATA\00", align 1
-@dynamic_size_global = common dso_local global i32* null, align 8
-@small_const_size_global = common dso_local global [8 x i32] zeroinitializer, align 16
-@large_const_size_global = common dso_local global [128 x i32] zeroinitializer, align 16
+@dynamic_size_global = dso_local global i32* null, align 8
+@small_const_size_global = dso_local global [8 x i32] zeroinitializer, align 16
+@large_const_size_global = dso_local global [128 x i32] zeroinitializer, align 16
 
 ; Function Attrs: noinline nounwind optnone uwtable
 define dso_local void @set(i32* %buf, i32 %sz) #0 {
@@ -31,7 +31,7 @@
 ; Function Attrs: nounwind
 declare dso_local i32 @rand() #1
 
-; Function Attrs: argmemonly nounwind willreturn
+; Function Attrs: argmemonly nounwind willreturn writeonly
 declare void @llvm.memset.p0i8.i64(i8* nocapture writeonly, i8, i64, i1 immarg) #2
 
 ; Function Attrs: noinline nounwind optnone uwtable
@@ -142,10 +142,10 @@
 ; Function Attrs: nounwind
 declare void @llvm.stackrestore(i8*) #4
 
-attributes #0 = { noinline nounwind optnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #1 = { nounwind "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
-attributes #2 = { argmemonly nounwind willreturn }
-attributes #3 = { nounwind readonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #0 = { noinline nounwind optnone uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #1 = { nounwind "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
+attributes #2 = { argmemonly nounwind willreturn writeonly }
+attributes #3 = { nounwind readonly "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "frame-pointer"="all" "less-precise-fpmad"="false" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }
 attributes #4 = { nounwind }
 attributes #5 = { nounwind readonly }
 
@@ -153,4 +153,4 @@
 !llvm.ident = !{!1}
 
 !0 = !{i32 1, !"wchar_size", i32 4}
-!1 = !{!"clang version 10.0.0-4ubuntu1 "}
+!1 = !{!"Ubuntu clang version 11.0.0-2~ubuntu20.04.1"}
```

</details>

Reply:
Here's the portion of the diff dealing with these globals:
```diff
--- 10.ll	2022-10-25 11:58:27.530693041 -0400
+++ 11.ll	2022-10-25 11:58:26.684687684 -0400
@@ -4,9 +4,9 @@
 target triple = "x86_64-pc-linux-gnu"
 
 @.str = private unnamed_addr constant [13 x i8] c"DYNAMIC_DATA\00", align 1
-@dynamic_size_global = common dso_local global i32* null, align 8
-@small_const_size_global = common dso_local global [8 x i32] zeroinitializer, align 16
-@large_const_size_global = common dso_local global [128 x i32] zeroinitializer, align 16
+@dynamic_size_global = dso_local global i32* null, align 8
+@small_const_size_global = dso_local global [8 x i32] zeroinitializer, align 16
+@large_const_size_global = dso_local global [128 x i32] zeroinitializer, align 16
```
The difference seems to be that in LLVM 11, these don't have an explicit `common` linkage.

Reply:
Let's diff everything from the FactGenerator:
```
pushd 10
sed_recurse() { ag -g '.*' -0 | xargs -0 sed -E -i "$@"; }
sed_recurse 's|10\.ll|11\.ll|g'
popd
for f in 10/*.csv.sorted; do diff -u "${f}" 11/$(basename ${f}); done
```

<details>
<summary>All diffs</summary>

```
--- 10/func_func_attr.csv.sorted	2022-10-25 14:08:08.639202435 -0400
+++ 11/func_func_attr.csv.sorted	2022-10-25 13:22:40.600448871 -0400
@@ -5,7 +5,7 @@
 </work/11.ll>:atoi	no-infs-fp-math=false
 </work/11.ll>:atoi	no-nans-fp-math=false
 </work/11.ll>:atoi	no-signed-zeros-fp-math=false
-</work/11.ll>:atoi	no-trapping-math=false
+</work/11.ll>:atoi	no-trapping-math=true
 </work/11.ll>:atoi	nounwind
 </work/11.ll>:atoi	readonly
 </work/11.ll>:atoi	stack-protector-buffer-size=8
@@ -20,7 +20,7 @@
 </work/11.ll>:getenv	no-infs-fp-math=false
 </work/11.ll>:getenv	no-nans-fp-math=false
 </work/11.ll>:getenv	no-signed-zeros-fp-math=false
-</work/11.ll>:getenv	no-trapping-math=false
+</work/11.ll>:getenv	no-trapping-math=true
 </work/11.ll>:getenv	nounwind
 </work/11.ll>:getenv	stack-protector-buffer-size=8
 </work/11.ll>:getenv	target-cpu=x86-64
@@ -43,7 +43,7 @@
 </work/11.ll>:main	no-jump-tables=false
 </work/11.ll>:main	no-nans-fp-math=false
 </work/11.ll>:main	no-signed-zeros-fp-math=false
-</work/11.ll>:main	no-trapping-math=false
+</work/11.ll>:main	no-trapping-math=true
 </work/11.ll>:main	nounwind
 </work/11.ll>:main	optnone
 </work/11.ll>:main	stack-protector-buffer-size=8
@@ -59,7 +59,7 @@
 </work/11.ll>:malloc	no-infs-fp-math=false
 </work/11.ll>:malloc	no-nans-fp-math=false
 </work/11.ll>:malloc	no-signed-zeros-fp-math=false
-</work/11.ll>:malloc	no-trapping-math=false
+</work/11.ll>:malloc	no-trapping-math=true
 </work/11.ll>:malloc	nounwind
 </work/11.ll>:malloc	stack-protector-buffer-size=8
 </work/11.ll>:malloc	target-cpu=x86-64
@@ -73,7 +73,7 @@
 </work/11.ll>:rand	no-infs-fp-math=false
 </work/11.ll>:rand	no-nans-fp-math=false
 </work/11.ll>:rand	no-signed-zeros-fp-math=false
-</work/11.ll>:rand	no-trapping-math=false
+</work/11.ll>:rand	no-trapping-math=true
 </work/11.ll>:rand	nounwind
 </work/11.ll>:rand	stack-protector-buffer-size=8
 </work/11.ll>:rand	target-cpu=x86-64
@@ -90,7 +90,7 @@
 </work/11.ll>:set	no-jump-tables=false
 </work/11.ll>:set	no-nans-fp-math=false
 </work/11.ll>:set	no-signed-zeros-fp-math=false
-</work/11.ll>:set	no-trapping-math=false
+</work/11.ll>:set	no-trapping-math=true
 </work/11.ll>:set	nounwind
 </work/11.ll>:set	optnone
 </work/11.ll>:set	stack-protector-buffer-size=8
--- 10/global_var_linkage.csv.sorted	2022-10-25 14:08:08.644202476 -0400
+++ 11/global_var_linkage.csv.sorted	2022-10-25 13:22:40.659449189 -0400
@@ -1,4 +1,4 @@
-</work/11.ll>:@dynamic_size_global	common
-</work/11.ll>:@large_const_size_global	common
-</work/11.ll>:@small_const_size_global	common
+</work/11.ll>:@dynamic_size_global	external
+</work/11.ll>:@large_const_size_global	external
+</work/11.ll>:@small_const_size_global	external
 </work/11.ll>:@.str	private
--- 10/target_dependent_attr.csv.sorted	2022-10-25 14:08:08.642202460 -0400
+++ 11/target_dependent_attr.csv.sorted	2022-10-25 13:22:40.925450623 -0400
@@ -44,12 +44,12 @@
 no-signed-zeros-fp-math=false
 no-signed-zeros-fp-math=false
 no-signed-zeros-fp-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
-no-trapping-math=false
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
+no-trapping-math=true
 stack-protector-buffer-size=8
 stack-protector-buffer-size=8
 stack-protector-buffer-size=8
```

</details>

Reply:
Ah, ok, here it is:
```diff
--- 10/global_var_linkage.csv.sorted	2022-10-25 14:08:08.644202476 -0400
+++ 11/global_var_linkage.csv.sorted	2022-10-25 13:22:40.659449189 -0400
@@ -1,4 +1,4 @@
-</work/11.ll>:@dynamic_size_global	common
-</work/11.ll>:@large_const_size_global	common
-</work/11.ll>:@small_const_size_global	common
+</work/11.ll>:@dynamic_size_global	external
+</work/11.ll>:@large_const_size_global	external
+</work/11.ll>:@small_const_size_global	external
```
Looking at the rules for global allocation sizes:
```
allocation_size(?globalAlloc, ?n) :-
  global_allocation_by_variable(?globalVar, ?globalAlloc),
  global_var_linkage(?globalVar, ?linkage),
  ! external_linkage_type(?linkage),
  static_allocation_type(?globalAlloc, ?type),
  type_has_size(?type, ?n),
  ! typeinfo_vtable(?globalVar, _).

// The sizes of external globals may not match the size of their types.
dynamically_sized_alloc(?alloc) :-
  external_linkage_type(?externalLinkage),
  global_var_linkage(?globalVar, ?externalLinkage),
  global_allocation_by_variable(?globalVar, ?alloc).
```
The size is no longer recorded because the FactGenerator thinks these have external linkage.

Here's the part of the FactGenerator that deals with this:

https://github.com/GaloisInc/cclyzerpp/blob/32e7e1a055e080600152d71704feb61f059b7f53/FactGenerator/src/globals.cpp

Reply:
The following diff:
```diff
diff --git a/test/c/allocation-sizes.c b/test/c/allocation-sizes.c
index ab70b89..d0d1add 100644
--- a/test/c/allocation-sizes.c
+++ b/test/c/allocation-sizes.c
@@ -7,9 +7,9 @@
 #define LARGE_CONST_SIZE 128
 #define SMALL_CONST_SIZE 8
 
-int small_const_size_global[SMALL_CONST_SIZE];
-int large_const_size_global[LARGE_CONST_SIZE];
-int *dynamic_size_global;
+int small_const_size_global[SMALL_CONST_SIZE] = {0};
+int large_const_size_global[LARGE_CONST_SIZE] = {0};
+int *dynamic_size_global = {0};
 
 void set(int *, int) __attribute__((noinline));
 void set(int *buf, int sz) {
```
makes cclyzer++ give the same results for LLVM 10 and 11, which is to *not* assign a size to these globals. That feels wrong! I think the bit in the above rules about external linkage is just incorrect - I wrote it to deal with `extern` globals that can be initialized elsewhere, but I think `extern` in C doesn't directly correspond to LLVM's notion of `extern` linkage.

In any case, the new result is OK relative to the LLVM 10 -> 11 upgrade.

Reply:
See #95, #96.